libs/blocks/merch/merch.js

```diff
@@ -37,16 +37,33 @@ export async function loadEntitlements() {
     fetch(entitlementsURL),
   ]).then(([{ default: getUserEntitlements }, mappings]) => {
     if (!mappings.ok) return [];
-    return Promise.all([getUserEntitlements(), mappings.json()]);
+    return Promise.all([getUserEntitlements({ params: [{ name: 'include', value: 'OFFER.PRODUCT_ARRANGEMENT' }], format: 'raw' }), mappings.json()]);
   });
   return loadEntitlements.promise;
 }
 
 async function getCheckoutAction(offers) {
-  const [{ arrangement_codes: aCodes } = {}, entitlementsMappings] = await loadEntitlements();
-  if (aCodes === undefined) return undefined;
+  let upgradeAction;
+  const [entitlements, entitlementsMappings] = await loadEntitlements();
+  const aCodes = entitlements?.map((offer) => offer.offer.product_arrangement_code);
   const [{ productArrangementCode }] = offers;
-  if (aCodes[productArrangementCode] === true) {
+  const [{ productArrangement }] = offers;
```

Comment:
As we discussed in a huddle:
```
async function getCheckoutAction(offers) {
  const [entitlements, entitlementsMappings] = await loadEntitlements();
  if (offers?.length) {
  const [{ productArrangementCode, productArrangement }] = offers || [];
  const { productFamily } = productArrangement?.productFamily || {};
  const upgradeOffer = await document.querySelector('.merch-offers.upgrade [data-wcs-osi]')?.onceSettled()
    .catch((e) => {
      window.lana.log('Failed to resolve an upgrade offer:', e);
      return undefined;
    });
  if (upgradeOffer && entitlements?.length && productFamily) {
    const { default: handleUpgradeOffer } = await import('./upgrade.js');
    const upgradeAction = handleUpgradeOffer(productFamily, upgradeOffer, entitlements);
   if (upgradeAction) return upgradeAction;
  }
 }
 
  const downloadFlow = false; // temporary disable
  if (downloadFlow === false) return undefined;
  const aCodes = entitlements?.map((offer) => offer.offer.product_arrangement_code);
  if (aCodes?.includes(productArrangementCode)) {
    const mapping = entitlementsMappings.data
      ?.find(({ [TITLE_PRODUCT_ARRANGEMENT_CODE]: code }) => code === productArrangementCode);
    if (!mapping) return undefined;
    const config = getConfig();
    const { locale: { ietf } } = config;
    const text = await replaceKey(mapping.CTA, config);
    const url = mapping[ietf] || localizeLink(mapping.Target);
    return { text, url };
  }
  return undefined;
}
```

Reply:
fixed

Reply:
Sorry, I forgot to remove one redundant checking, instead of:
 ```const [{ productArrangementCode, productArrangement }] = offers || [];```
 should be:
  ```const [{ productArrangementCode, productArrangement }] = offers;```

Reply:
yep, this is what i have atm